tools/autograd/gen_variable_factories.py

```diff
@@ -47,29 +84,69 @@ def gen_variable_factories(out, declarations, template_path, disable_autograd=Fa
           CodeTemplate.from_file(template_path + "/variable_factories.h"),
           {"function_definitions": function_definitions})
 
-
+def replace_dtype_nullprt(actuals):
+    replaced = actuals[:]
+    index = actuals.index('at::typeMetaToScalarType(options.dtype())')
+    replaced[index] = 'c10::nullopt'
+    return replaced
+    
 def process_function(decl, has_tensor_options, disable_autograd):
     formals = []
     actuals = []
     for argument in decl["arguments"]:
         type = fully_qualified_type(argument["type"])
+        
         default = " = {}".format(argument["default"]) if "default" in argument else ""
+        if "default" in argument:
+            if argument["default"] == False or argument["default"] == True:
+                default = default.lower()
+
         formals.append("{} {}{}".format(type, argument["name"], default))
+
         actual = argument["name"]
         if argument["simple_type"] == "TensorOptions":
             actual = "at::TensorOptions({})".format(actual)
         actuals.append(actual)
-    requires_grad = "options.requires_grad()" if has_tensor_options else "false"
+
+    requires_grad = "false"
     if decl['name'].endswith('_like') and not has_tensor_options:
         # Insert TensorOptions before MemoryFormat
-        actuals.insert(-1, '{}.options()'.format(actuals[0]))
-
+        actuals.insert(-1, 'at::typeMetaToScalarType({}.options().dtype())'.format(actuals[0]))
```

Comment:
why is the spelled-out version ScalarType based and not TypeMeta based?  (honest question).